wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -121,8 +132,38 @@ public Schema getOutputSchema(SchemaResolutionContext context) {
         .map(
           field -> {
             try {
-              return field.getName().equals(col) ?
-                Schema.Field.of(col, ColumnConverter.getSchemaForType(type, scale)) : field;
+              if (field.getName().equals(col)) {
+                Schema intermediateSchema = field.getSchema().getNonNullable();
```

Comment:
Can rename to inputSchema

Reply:
Call it "fieldSchema" then, intermediateSchema is a bit confusing

Reply:
Done